go-controller/pkg/util/net.go

```diff
@@ -110,3 +110,9 @@ func GetNodeWellKnownAddresses(subnet *net.IPNet) (*net.IPNet, *net.IPNet) {
 func JoinHostPortInt32(host string, port int32) string {
 	return net.JoinHostPort(host, strconv.Itoa(int(port)))
 }
+
+// IPAddrToHWAddr takes the four octets of IPv4 address (aa.bb.cc.dd, for example) and uses them in creating
+// a MAC address (00:00:aa:bb:cc:dd)
+func IPAddrToHWAddr(ip net.IP) string {
+	return fmt.Sprintf("00:00:%02X:%02X:%02X:%02X", ip[0], ip[1], ip[2], ip[3])
```

Comment:
@girishmg we should use a MAC prefix in the safe range I think, like how CNI does it (first two bytes are 0A:58).

Reply:
(the "safe range", aka "locally-assigned" is any address where `(mac[0] & 0x3) == 0x2`)

Reply:
@danwinship right and octet 0A satisfies that definition.